wrappers/services.go

```diff
@@ -811,15 +811,16 @@ func StopServices(apps []*snap.AppInfo, flags *StopServicesFlags, reason snap.Se
 	return nil
 }
 
-// ServicesEnableState returns a map of service names from the given snap,
-// together with their enable/disable status.
-func ServicesEnableState(s *snap.Info, inter interacter) (map[string]bool, error) {
+// ServicesEnableState takes a map of application names -> application info
+// and returns a map of service names extracted from the given map, together
+// with their enable/disable status.
+func ServicesEnableState(appInfos map[string]*snap.AppInfo, inter interacter) (map[string]bool, error) {
```

Comment:
Is there a deep reason for changing the signature of this function (and QueryDisabledServices below) from taking snap.Info to appInfos map? We are passing all apps anyway right?

Reply:
Well, it's to make it easier to reuse this function: while the original version took a `snap.Info` as parameter, the only thing it needs to know is the list of the service names which are inside it.

Reply:
I reverted the change, since after refactoring the code to delay the computation of the disabled services I did indeed have a `snap.Info` object to pass as a parameter. But in the general, it's always better if a function takes as parameters the minimum amount of information it needs to performs its task.

Reply:
Thanks. Yes, this is generally true, but snap.Info is a bit special because we often need to introduce logic/conditionals based on some other attributes of the snap, and it is annoying to then have to refactor a lot of unrelated code to have snap.Info passed again. It's also a way of saying that the function works on services of the given snap (also it is not a requirement for correctness of this function).  It also keeps this function consistent with other helpers from wrappers.